File: Clase__75__15-10__Room/EjercicioRoom/app/src/main/java/com/example/ejercicioroom/fragmentos/DetailsFragment.java
package com.example.ejercicioroom.fragmentos;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.ejercicioroom.databinding.FragmentDetailsBinding;
import com.example.ejercicioroom.model.ProductoEntity;
import com.example.ejercicioroom.viewmodel.ProductoViewModel;


public class DetailsFragment extends Fragment {
    FragmentDetailsBinding binding;
    ProductoViewModel viewModel;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentDetailsBinding.inflate(inflater, container, false);

       // com.example.ejercicioroom.model.Producto producto = DetailsFragmentArgs.fromBundle(getArguments()).getProducto();

        viewModel = new ViewModelProvider(this).get(ProductoViewModel.class);
        ProductoEntity producto = viewModel.getCurrentProduct().getValue();

        //Log.i("prueba", producto.getMarca().toString());
//
        binding.tvNombre.setText(producto.getModelo());
//
//        binding.tvPrecio.setText("$" + producto.getPrecio());



        return binding.getRoot();
    }
}

File: Clase__83__27-10__Kotlin&Android/Pizzeria Juanito/app/src/main/java/com/example/pizzeriajuanito/fragments/ui/DetailsFragment.kt
package com.example.pizzeriajuanito.fragments.ui

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.activityViewModels
import com.example.pizzeriajuanito.R
import com.example.pizzeriajuanito.databinding.FragmentDetailsBinding
import com.example.pizzeriajuanito.viewmodel.PizzaViewModel


class DetailsFragment : Fragment() {
    private lateinit var binding: FragmentDetailsBinding
    val viewModel: PizzaViewModel by activityViewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentDetailsBinding.inflate(layoutInflater, container, false)

        var total = viewModel.pizza.total
        var cliente = viewModel.pizza.cliente

        binding.textView.text = total.toString()
        binding.textView2.text = cliente.toString()



        return binding.root

    }


}